### DISCUSSION TITLE
Question about the number density

### DISCUSSION THREAD

**Discussion by smallrainbow** [epochpic/epoch#141] [Q&A] [answered]
Dear developers,

Thanks for your notice. It's good to know that epoch has been moved to github.

I have a question about the input deck. When I use epoch to read an external file to set the initial particle number density distribution, does the epoch defaults to set the same weight of each super-particle, and have a spatial-distribution of number of particles per cell? Or does the epoch uniformly distribute the super-particles throughout the space, but change the weight of each particle to fit the given number density profile?

If the former is not the default, how to modify the Makefile and other files to use the former one, thank you!

**Comment by smallrainbow** [comment]
In short, how can I let epoch **keep the weight of each particle the same** when initializing structured density profiles, and **only change the number of particles in each grid cell** to satisfy a given structured density profile (e.g., on Page 86-87 in the user's manual)?

Thank you very much!

**Comment by TomGoffrey** [comment, marked as answer]
I believe the behaviour you want would be enabled by the `PER_SPECIES_WEIGHT` compilation option, so you could compile with `make COMPILER=gfortran DEFINES=-DPER_SPECIES_WEIGHT`. Remember to `make clean` beforehand.

**Reply by smallrainbow** [reply]
@TomGoffrey 

Thank you very much!

I have tried 2 versions of Epoch.

(1) epoch-v4.17.13 is fine. The compilation is finished without any error or warning. So it seems good.

(2) epoch-v4.16.6 reports some errors as follows, please see below.

mpiifort -c -O3 -g -stand f03 -I../SDF/FORTRAN/include -module obj -o obj/diagnostics.o -DPER_SPECIES_WEIGHT -D_COMMIT='"v4.16.6-0-g77f7e800-clean"' -D_DATE=1613467128 -D_MACHINE='"mn01"' -Isrc/include src/io/diagnostics.F90
src/io/diagnostics.F90(2451): error #6460: This is not a field name that is defined in the encompassing structure.   [WEIGHT]
        .AND. current%weight < sub%weight_min) &
----------------------^
compilation aborted for src/io/diagnostics.F90 (code 1)
make[1]: *** [diagnostics.o] Error 1
make[1]: Leaving directory `/gpfs/home/yangzw/epoch-4.16.6_parallel_shock/epoch2d'
make: *** [commit] Error 2

**Reply by TomGoffrey** [reply]
Yes, it looks like there was a bug in 4.16.6, however, as that version is almost two years old, I don't recommend its use.
